### DISCUSSION TITLE
Turn Non-static C# Function into SKFunction

### DISCUSSION THREAD

**Discussion by HazyFish** [microsoft/semantic-kernel#3363] [Q&A] [answered]
Is it possible to turn a non-static C# function into a Semantic Kernel native function?

We need to use Dependency Injection.

**Comment by stephentoub** [comment, marked as answer]
> Is it possible to turn a non-static C# function into a Semantic Kernel native function?

Yes, you're looking for SKFunction.FromNativeMethod/FromNativeFunction.
